File: venus/src/model/interface/spot/spotSearchRequestDto.ts
import { SpotSortType } from "../../enum/spot/spotSortType";
import { SpotRatingFilterType } from "../../enum/spot/spotRatingFilterType";

export interface SpotSearchRequestDto {
    city?: string;
    tags?: string[];
    sort: SpotSortType;
    filter: SpotRatingFilterType;
    userLongitude?: number;
    userLatitude?: number;
}

File: vulcanus/src/main/java/com/merkury/vulcanus/model/dtos/spot/SpotSearchRequestDto.java
package com.merkury.vulcanus.model.dtos.spot;

import com.merkury.vulcanus.model.enums.SpotRatingFilterType;
import com.merkury.vulcanus.model.enums.SpotSortType;

import java.util.List;

public record SpotSearchRequestDto(String city,
                                   List<String> tags,
                                   SpotSortType sort,
                                   SpotRatingFilterType filter) {
}
